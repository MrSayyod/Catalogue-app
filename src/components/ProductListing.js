import React, { useEffect } from "react";
import axios from "axios";
import { useSelector } from "react-redux";
import ProductComponent from "./ProductComponent";
import { useDispatch } from "react-redux";
import { setProducts } from "../actions/productActions";
import { filterProduct } from "../actions/productActions";
import { CategoryFilter } from "./CategoryFilter";

const ProductListing = () => {
  const products = useSelector((state) => state);
  const all = products.allProducts.products;
  const dispatch = useDispatch();
  const fetchProducts = async () => {
    const response = await axios
      .get("https://fakestoreapi.com/products")
      .catch((error) => {
        console.log("Error", error);
      });
    dispatch(setProducts(response.data));
  };
  useEffect(() => {
    fetchProducts();
  }, []);
  const handleFilterChange = (e) => {
    dispatch(filterProduct(e.target.value));
  };
  const productsDisplay = () => {
    if (products.filter === "ALL") {
      return products.allProducts.products;
    }
    return products.allProducts.products.filter(
      (product) => product.category === products.filter
    );
  };
  let content;
  let filteredContent;
  if (products.filter === "ALL") {
    content = <ProductComponent items={all} />;
  } else {
    filteredContent = all.filter(
      (category) => category.category === products.filter
    );
    console.log(filteredContent);
    content = <ProductComponent items={filteredContent} />;
  }
  return (
    <div className="menu_pdls">
      <div className="category_filter">
        <CategoryFilter
          handleFilterChange={handleFilterChange}
          className="categoryDropdown"
        />
      </div>
      <div className="container_pdls"></div>
      <section>{content}</section>
    </div>
  );
};

export default ProductListing;
