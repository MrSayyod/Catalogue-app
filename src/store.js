import { createStore } from 'redux';
import reducers from './reducers';

const store = createStore(
  reducers,
  {},
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
);
/* eslint no-underscore-dangle: ["error", { "allow": ["__REDUX_DEVTOOLS_EXTENSION__"] }] */
export default store;
