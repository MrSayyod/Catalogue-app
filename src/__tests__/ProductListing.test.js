import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import ProductListing from '../components/ProductListing';
import store from '../store';

it('Should return an array of anime list', () => {
  const getData = async () => {
    const response = await fetch('https://fakestoreapi.com/products');
    const myData = await response.json();
    return myData.allProducts;
  };
  expect(getData()).not.toEqual(null);
});

it('Expect to render Anime List component', () => {
  const result = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <ProductListing />
      </BrowserRouter>
    </Provider>,
  ).toJSON;
  expect(result).toMatchSnapshot();
});
