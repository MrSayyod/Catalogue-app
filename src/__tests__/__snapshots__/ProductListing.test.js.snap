// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render Anime List component 1`] = `[Function]`;
